pkg/apis/serving/v1alpha1/inference_graph.go

```diff
@@ -335,3 +337,25 @@ type InferenceGraphList struct {
 func init() {
 	SchemeBuilder.Register(&InferenceGraph{}, &InferenceGraphList{})
 }
+
+func (ig *InferenceGraph) HasAuthEnabled() bool {
+	return ig.GetAnnotations()[constants.EnableAuthAnnotationKey] == "true"
+}
+
+// ResolveServiceAccountName returns the ServiceAccount name to use for the InferenceGraph pod.
+// If the user has customized the name of the ServiceAccount, the custom name is used. If there is no
+// custom ServiceAccount the returned name depends on whether the IG is auth-protected:
+//   - If the IG is *not* auth-protected, `default` is returned (backwards compatibility)
+//   - If the IG is *not* auth-protected, the returned value is the name of the IG with `-auth-verifier` suffix
+//     in a try to stay close to principle of least privilege.
+func (ig *InferenceGraph) ResolveServiceAccountName() string {
+	if len(ig.Spec.ServiceAccountName) != 0 {
+		return ig.Spec.ServiceAccountName
```

Comment:
At this point, I suppose the validation to make sure that the SA exists was already done when the pod was created, right?

Reply:
I don't think there is a validation for this. It is probably user responsibility doing that, if they are customizing the SA.

Reply:
yeah, agreed.